src/libraries/System.Utf8String.Experimental/src/System/Runtime/Intrinsics/Intrinsics.Shims.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Runtime.Intrinsics
+{
+    internal static class Vector128
+    {
+        public static Vector128<short> Create(short value) => throw new PlatformNotSupportedException();
+        public static Vector128<ushort> Create(ushort value) => throw new PlatformNotSupportedException();
+        public static Vector128<ulong> CreateScalarUnsafe(ulong value) => throw new PlatformNotSupportedException();
+        public static Vector128<byte> AsByte<T>(this Vector128<T> vector) where T : struct => throw new PlatformNotSupportedException();
+        public static Vector128<short> AsInt16<T>(this Vector128<T> vector) where T : struct => throw new PlatformNotSupportedException();
+        public static Vector128<ushort> AsUInt16<T>(this Vector128<T> vector) where T : struct => throw new PlatformNotSupportedException();
+        public static Vector128<uint> AsUInt32<T>(this Vector128<T> vector) where T : struct => throw new PlatformNotSupportedException();
+        public static Vector128<ulong> AsUInt64<T>(this Vector128<T> vector) where T : struct => throw new PlatformNotSupportedException();
+        public static T GetElement<T>(this Vector128<T> vector, int index) where T : struct => throw new PlatformNotSupportedException();
+    }
+    internal readonly struct Vector128<T>
+        where T : struct
+    {
+        public static Vector128<T> Zero => throw new PlatformNotSupportedException();
+        public static int Count => throw new PlatformNotSupportedException();
+    }
+}
+
+namespace System.Runtime.Intrinsics.X86
+{
+    internal abstract class Bmi1
+    {
+        public abstract class X64
+        {
+            public static bool IsSupported { get; } = false;
```

Comment:
For all of the `IsSupported` checks, I'd change these to be _const fields_ instead of _static readonly properties_. This will allow the C# compiler to remove the relevant branches at compile time, so we don't have to worry about moving these optimizations to the JIT.

This is likely to result in a bunch of [CS0162](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0162) warnings. What I'd do there is modify the _System.Utf8String.Experimental.csproj_ itself and suppress that warning at the .csproj level, but only when targeting netstandard2.0.

Reply:
I was contemplating including the current PlatformNotSupported files, ex.

https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/X86/Sse42.PlatformNotSupported.cs

That way this file doesn't need to be maintained. It is more IL to include in the package, and we couldn't do this `const fields` trick. Do you have a preference on which way to go here?

Reply:
I ended up taking your suggestion and turning IsSupported into const fields.